### PR TITLE
Fix gradient check in `convert_points_from_homogeneous`

### LINKED ISSUE #893: convert_points_to_homogeneous gradcheck error for z == 0
## 🐛 Bug

The function `convert_points_to_homogeneous` raises an error in gradcheck when the z value is absolute zero.

## To Reproduce

```python
    def test_gradcheck_zvec_zeros(self, device, dtype):
        # generate input data
        points_h = torch.tensor([
            [1., 2., 0.],
            [0., 1., 0.1],
            [2., 1., 0.1],
        ], device=device, dtype=dtype)

        # evaluate function gradient
        points_h = tensor_to_gradcheck_var(points_h)  # to var
        assert gradcheck(kornia.convert_points_from_homogeneous, (points_h,),
                         raise_exception=True)
```

## Additional context

https://github.com/kornia/kornia/blob/0a60ac82f8afc6a95e42b1d831659db51b7509a3/test/geometry/test_conversions.py#L579-L591

### PR DESCRIPTION
**Pull request by manurare**
* `convert_points_from_homogeneous` avoids dividing by the last component (`z`) if z<=1e-8 while Pytorch `gradcheck` computes finite differences with eps=1e-6.

* By setting `gradcheck`'s eps to 1e-8, z==0 evaluates correctly. Then, the numerical and analytical gradient match and the test passes.

#### Changes

Fixes #893 

#### Type of change
- [x] 🐞 Bug fix (non-breaking change which fixes an issue)

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

**Comment by Meggison** [comment]
@edgarriba seems like a reasonable fix

**Comment by manurare** [comment]
> can we fix CI failure before merging

Is this something I should do myself?

**Comment by edgarriba** [comment]
@manurare can you rebase from `main` branch?

**Comment by manurare** [comment]
My fix branch is rebased on top of the most recent kornia main. Do I need to create a new PR with my `main` branch rebased from my fix branch?

**Comment by Meggison** [comment]
@manurare Seems fine now. @edgarriba should the precision issue be covered in a seperate PR?

**Comment by edgarriba** [comment]
> @manurare Seems fine now. @edgarriba should the precision issue be covered in a seperate PR?

yes, if you can check that one
